plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -763,12 +762,13 @@ enterComputeCek = computeCek
     -- s , constr I V0 ... Vj-1 _ (Tj+1 ... Tn, ρ) ◅ Vj  ↦  s , constr i V0 ... Vj _ (Tj+2... Tn, ρ)  ; ρ ▻ Tj+1
     returnCek (FrameConstr env i todo done ctx) e = do
         let done' = ConsStack e done
-        case todo of
-          (next : todo') -> computeCek (FrameConstr env i todo' done' ctx) env next
-          _              -> returnCek ctx $ VConstr i done'
+        case V.uncons todo of
+          Just (next, todo') -> computeCek (FrameConstr env i todo' done' ctx) env next
+          Nothing            -> returnCek ctx $ VConstr i done'
     -- s , case _ (C0 ... CN, ρ) ◅ constr i V1 .. Vm  ↦  s , [_ V1 ... Vm] ; ρ ▻ Ci
     returnCek (FrameCases env cs ctx) e = case e of
-        (VConstr i args) -> case cs ^? wix i of
+        -- TODO: handle word/int conversion better
```

Comment:
Also unsure what to do here. Check if the `Integer` is in range, otherwise assume it's a miss?

Reply:
It's _probably_ fine, because you're only going to get a negative integer if an overflow occurs and that will be caught by `(V.?!)`, but this really does need some extensive manual testing.

You could also do what I did in the `SmallArray` PR:

```
        VConstr i args
            | i < unsafeCoerce (sizeofSmallArray cs) ->
                computeCek (transferArgStack args ctx) env . indexSmallArray cs $ unsafeCoerce i
```

where `indexSmallArray` is unsafe lookup and `sizeofSmallArray` stupidly returns the length of the array rather than its byte size (man it pisses me off that they literally gave up on the usual camelCasing and stole the name of a C primitive just to assign it a completely different meaning). The logic here is

- we know that the length of the array is a non-negative `Int`, so it's safe to `unsafeCoerce` it to a `Word64`
- once we know that `i` is within the array bounds, it's safe to `unsafeCoerce` it to an `Int`, unless the array is somehow larger than `fromIntegral (maxBound :: Int) :: Word64`, which isn't something we could deserialize

Please note that I absolutely do not propose to have `unsafeCoerce` in there, just explaining the reasoning. Replacing both with `fromIntegral` should do the job just fine.

Reply:
> It's probably fine, because you're only going to get a negative integer if an overflow occurs and that will be caught by (V.?!), but this really does need some extensive manual testing.

Yeah no, you can still wrap around to positive again.

I'll add some kind of bounds check.

Reply:
Okay, so it can in fact never trigger, because `i` is a `Word64` and the largest `Word64` wraps to only `-1` as an `Int64`, so can't produce an "apparently good" value. I've left the check in out of caution.

I wanted to write a conformance test for this, but I was blocked by the fact that the parser doesn't do range checks, so in fact already wraps a "too large" number around problematically :joy: but that's a parser bug.